### PR TITLE
MGMT-2282 Create a random name for the initial iso job

### PR DESCRIPTION
**Pull request by carbonin**
Previously if we failed to clean up one of these jobs we would not
be able to re-deploy

### CODE REVIEW

**Comment by romfreiman** [comment]
I think it's irrelevant, no?

**Comment by carbonin** [comment]
No, the last deploy to integration was failing until I removed the old job manually.

Logs from the failing pod looked like this:
```
time="2020-09-25T20:31:48Z" level=info msg="Creating job livecd.iso" func="github.com/openshift/assisted-service/pkg/job.(*kubeJob).UploadBaseISO" file="/go/src/github.com/openshift/origin/pkg/job/job.go:292" go-id=756 pkg=k8s-job-wrapper request_id=
time="2020-09-25T20:31:48Z" level=error msg="failed to create image job" func="github.com/openshift/assisted-service/pkg/job.(*kubeJob).UploadBaseISO" file="/go/src/github.com/openshift/origin/pkg/job/job.go:294" error="jobs.batch \"livecd.iso\" already exists" go-id=756 pkg=k8s-job-wrapper request_id=
time="2020-09-25T20:31:48Z" level=fatal msg="Failed to upload base imagejobs.batch \"livecd.iso\" already exists" func=main.main.func1 file="/go/src/github.com/openshift/origin/cmd/main.go:292"
```

**Review by avishayt — APPROVED** [review]

**Review by eranco74 — COMMENTED** [review]
/approve

File: pkg/job/job.go [hunk 1]
```diff
@@ -288,20 +289,35 @@ func (k *kubeJob) uploadImageJob(jobName, imageName string) *batch.Job {
 func (k *kubeJob) UploadBaseISO() error {
 	ctx := context.Background()
 	log := logutil.FromContext(ctx, k.log)
+	jobName, err := uploadBaseISOJobName()
```
Comment (mhrivnak) [inline pkg/job/job.go:292]: You may just want to use this instead: https://github.com/kubernetes/apiserver/blob/v0.19.2/pkg/storage/names/generate.go#L49

Comment (mhrivnak) [reply pkg/job/job.go:292]: Or perhaps you can just set the Job's `GenerateName` field instead of the `Name` field, and let the api server do this work.

**Comment by mhrivnak** [comment]
/hold
just to consider the options I suggested

**Comment by carbonin** [comment]
/unhold

Went with the `GenerateName` route. Thanks @mhrivnak

**Review by avishayt — APPROVED** [review]

**Pull request merged**